src/channel.cr

```diff
@@ -15,12 +16,55 @@ require "fiber"
 # channel.receive # => 0
 # channel.receive # => 1
 # ```
-abstract class Channel(T)
-  module SelectAction
-    abstract def ready?
-    abstract def execute
-    abstract def wait
+class Channel(T)
+  @lock = Crystal::SpinLock.new
+  @queue : Deque(T)?
+
+  module NotReady
+    extend self
```

Comment:
Is this `extend self` needed here?

Reply:
It's a way to create unit types. Same as empty structs, but you don't need to put a `T.new` to refer the unit value. https://twitter.com/bcardiff/status/1157007260985495552

Reply:
Oooooh... I see. But you can also use `is_a?(NotReady.class)`, but I guess it's easier to use the `extend self` trick. Nice!